### ISSUE TITLE
Numbat running for two samples separately but crashes when I merge the two and perform the same analysis again

### ISSUE THREAD

**Issue by quail768** [kharchenkolab/numbat#203]
Hey Teng,

Really amazing tool!
I am trying to run Numbat on two samples from the same patient.

 I have run the Pileup and Phase script for both samples individually and Numbat seems to work when I run them individually(2 separate Numbat runs for 2 separate samples)

When I merge the two samples and try to run Numbat on the merged sample, I get the following error:

```
INFO [2024-08-26 11:27:34] 
numbat version: 1.4.1
scistreer version: 1.2.0
hahmmr version: 1.0.0
Running under parameters:
t = 1e-07
alpha = 1e-04
gamma = 20
min_cells = 40
init_k = 10
max_cost = 12468.9
n_cut = 6
max_iter = 2
max_nni = 100
min_depth = 0
use_loh = auto
segs_loh = None
call_clonal_loh = FALSE
segs_consensus_fix = Given
multi_allelic = TRUE
min_LLR = 5
min_overlap = 0.45
max_entropy = 0.7
skip_nj = FALSE
diploid_chroms = None
ncores = 15
ncores_nni = 15
common_diploid = TRUE
tau = 0.3
check_convergence = TRUE
plot = TRUEa
genome = hg38
Input metrics:
41563 cells
INFO [2024-08-26 11:27:35] Mem used: 10.8Gb
INFO [2024-08-26 11:28:06] Approximating initial clusters using smoothed expression ..
INFO [2024-08-26 11:28:08] Mem used: 10.8Gb
INFO [2024-08-26 13:51:18] running hclust...
INFO [2024-08-26 14:05:29] Iteration 1
INFO [2024-08-26 14:05:33] Mem used: 35.7Gb
INFO [2024-08-26 14:06:03] Expression noise level (MSE): medium (1.3). 
INFO [2024-08-26 14:06:03] Using fixed consensus CNVs
INFO [2024-08-26 14:07:56] Running HMMs on 10 cell groups..
ERROR [2024-08-26 14:08:15] No segments containing more than 10 genes for CHROM 21.
ERROR [2024-08-26 14:08:15] Error in smooth_segs(., min_genes = min_genes) : 
No segments containing more than 10 genes for CHROM 21.

```

I have re-ran the Pileup and Phase for both samples together as instructed in the manual. 
Numbat runs for the same samples separately so I'm assuming that there are in-fact, genes for CHROM21

Any thoughts on what could be going wrong or something I could change?

Thanks for the help!

**Comment by quail768** [comment]
Hey @teng-gao . 
Thanks for the reply.
Sorry if I was unclear. 

Like I said earlier, I did run pileup and phase jointly for both samples that I planed to use for the combined Numbat run. From that single pileup and phase run, I have multiple allele_df files, all of which were included in the main Numbat run. I have run the multi-sample mode before with no problems.

Any thoughts?

**Comment by boneoncology** [comment]
I encountered the same issue, there is nothing wrong with seperate sample running, but error came out when I merge them together. I followed the guideline to make unique prefixed-sample barcode for each reads in merged BAM to run pile up phase.

**Comment by teng-gao** [comment]
Hi @boneoncology could you share some example data to reproduce the error? Doesn't have to be the full dataset. Thanks

**Comment by boneoncology** [comment]
> Hi [@boneoncology](https://github.com/boneoncology) could you share some example data (expression matrix and allele df) to reproduce the error? Doesn't have to be the full dataset. Thanks

> out = run_numbat(
+   count_mat_os, # gene x cell integer UMI count matrix 
+   ref_hca, 
+   df_allele_filtered,
+   genome = "hg38",
+   t = 1e-5,
+   ncores = 32,
+   plot = TRUE,
+   out_dir = "/home/MyPassport/sda/numbat/OS_CELL",min_genes = 1)


numbat version: 1.4.2
scistreer version: 1.2.0
hahmmr version: 1.0.0
Running under parameters:
t = 1e-05
alpha = 1e-04
gamma = 20
min_cells = 50
init_k = 3
max_cost = 5841.6
n_cut = 0
max_iter = 2
max_nni = 100
min_depth = 0
use_loh = auto
segs_loh = None
call_clonal_loh = FALSE
segs_consensus_fix = None
multi_allelic = TRUE
min_LLR = 5
min_overlap = 0.45
max_entropy = 0.5
skip_nj = FALSE
diploid_chroms = None
ncores = 32
ncores_nni = 32
common_diploid = TRUE
tau = 0.3
check_convergence = FALSE
plot = TRUE
genome = hg38
Input metrics:
19472 cells
Mem used: 26.6Gb
Approximating initial clusters using smoothed expression ..
Mem used: 26.6Gb
number of genes left: 11109
running hclust...
                                                                                                                                     
Iteration 1
Mem used: 26.6Gb
Expression noise level (MSE): high (1.9). Consider using a custom expression reference profile.
Running HMMs on 5 cell groups..
错误于find_common_diploid(bulks, gamma = gamma, alpha = alpha, ncores = ncores): 
  Error in smooth_segs(., min_genes = min_genes) : 
  No segments containing more than 10 genes for CHROM 18.
此外: 警告信息:
1: In asMethod(object) :
  sparse->dense coercion: allocating vector of size 2.0 GiB
2: In asMethod(object) :
  sparse->dense coercion: allocating vector of size 1.6 GiB
3: In mclapply(bulks %>% split(.$sample), mc.cores = ncores, function(bulk) { :
  预定内核5, 2的用户代码中出了错，所有作业值都将受到影响
> count_mat_os[1:10,1:10]
           BS1_AAACCCACAATGAAAC-1 BS1_AAACCCAGTGCACATT-1 BS1_AAACCCATCCGCTGTT-1 BS1_AAACCCATCTCATAGG-1 BS1_AAACCCATCTTTGGAG-1 BS1_AAACGAAAGGGCTTCC-1 BS1_AAACGAACACAGTCAT-1 BS1_AAACGAACACCTTCCA-1
AL627309.1                      0                      0                      0                      0                      0                      0                      0                      0
AL627309.5                      0                      0                      0                      0                      0                      0                      0                      0
AP006222.2                      0                      0                      0                      0                      0                      0                      0                      0
AL669831.2                      0                      0                      0                      0                      0                      0                      0                      0
LINC01409                       0                      0                      1                      0                      0                      1                      0                      1
FAM87B                          0                      0                      0                      0                      0                      0                      0                      0
LINC01128                       0                      0                      0                      1                      1                      0                      0                      0
LINC00115                       0                      0                      0                      0                      0                      0                      0                      1
FAM41C                          0                      0                      0                      0                      0                      0                      0                      0
AL645608.6                      1                      0                      0                      1                      0                      0                      0                      0
           BS1_AAACGAACACGATTCA-1 BS1_AAACGAAGTGTTTACG-1
AL627309.1                      0                      0
AL627309.5                      0                      0
AP006222.2                      0                      0
AL669831.2                      0                      0
LINC01409                       1                      0
FAM87B                          0                      0
LINC01128                       0                      0
LINC00115                       0                      1
FAM41C                          0                      0
AL645608.6                      1                      1
> head(df_allele_filtered)
                      cell       snp_id CHROM    POS cM REF ALT AD DP  GT gene
1   BS3_GACGTTAGTGACTGTT-1 1_263722_C_G     1 263722  0   C   G  0  1 1|0     
4  BS12_AACACACCAAACCACT-1 1_779047_G_A     1 779047  0   G   A  1  1 0|1     
5  BS12_AACCCAACACCTTCGT-1 1_779047_G_A     1 779047  0   G   A  0  1 0|1     
7  BS12_ACCAAACCACTATCCC-1 1_779047_G_A     1 779047  0   G   A  1  1 0|1     
10 BS12_CAACCTCGTGCATTTG-1 1_779047_G_A     1 779047  0   G   A  1  1 0|1     
12 BS12_CATCCACCAGTGGTGA-1 1_779047_G_A     1 779047  0   G   A  0  1 0|1
> table(df_allele_filtered$CHROM)

      1      10      11      12      13      14      15      16      17      18      19       2      20      21      22       3       4       5       6 
1724619  908936 1024254  952319  496182  713317  638576  584814  884011  393062  701244 1449681  451849  269327  313893 1430917 1309189 1083317 1355959 
      7       8       9 
1250710 1086883 1039017 
Hi  the pileup and phase run for 14 samples merged BAM was successful. I tried to set min_genes = 1, 5, 10, etc, the error came out like 10 genes for CHROM 18 or 5 genes for CHROM 21. Everything was OK with single sample run, I got all the final trees, subclones and graphs.
